File: maximum-profit-in-job-scheduling.nim
import std/algorithm

type cintA = ptr UncheckedArray[cint]

proc jobScheduling(startTime: cintA, n: int, endTime: cintA, u0: cint, profit: cintA, u1: cint): cint {.exportc.} =
  var
    a = newSeq[(cint, cint, cint)](n)
    dp = newSeq[cint](n)
  for i in 0..<n:
    a[i] = (startTime[i], endTime[i], profit[i])
  a.s

File: maximum-profit-in-job-scheduling.rs
// Maximum Profit in Job Scheduling
impl Solution {
    pub fn job_scheduling(start_time: Vec<i32>, end_time: Vec<i32>, profit: Vec<i32>) -> i32 {
        let n = start_time.len();
        let mut a = vec![(0,0,0); n];
        for i in 0..n { a[i] = (start_time[i], end_time[i], pr